include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -131,34 +130,6 @@ inline void enrich_assign(Operations& operations, Turns& turns,
             return next_turn.operations[operations[next_index].operation_index];
         };
 
-        if (check_consecutive_turns
-            && indexed.turn_index == operations[next_index].turn_index
-            && op.seg_id == next_operation().seg_id)
-        {
-            // If the two operations on the same turn are ordered consecutively,
-            // and they are on the same segment, then the turn where to travel to should
-            // be considered one further. Therefore next is increased.
-            //
-            // It often happens in buffer, in these configurations:
-            // +---->--+
-            // |       |
-            // |   +->-*---->
-            // |   |   |
-            // ^   +-<-+
-            // If the next index is not corrected, the small rectangle
-            // will be kept in the output.
-
-            // This is a normal situation and occurs, for example, in every concave bend.
-            // In general it should always travel from turn to next turn.
-            // Only in some circumstances traveling to the same turn is necessary, for example
-            // if there is only one turn in the outer ring.
-            //
-            // (For dissolve this is not done, turn_index is often
-            // the same for two consecutive operations - but the conditions are changed
-            // and this should be verified again)
-            next_index = advance(next_index);
```

Comment:
This functionality was active - but it is long obsolete and now handled in another way (and will change in next PR)